src/iqm/qiskit_iqm/iqm_move_layout.py

```diff
@@ -11,34 +11,34 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""A layout algorithm that generates an initial layout for a quantum circuit that is
-valid on the quantum architecture specification of the given IQM backend."""
+"""Generate an initial layout for a quantum circuit that is
+valid on the quantum architecture specification of the given backend."""
+from typing import Optional, Union
+
 from qiskit import QuantumCircuit
 from qiskit.dagcircuit import DAGCircuit
 from qiskit.transpiler import PassManager, TranspilerError
+from qiskit.transpiler.layout import Layout
 from qiskit.transpiler.passes import TrivialLayout
 
+from iqm.qiskit_iqm.iqm_backend import IQMTarget
 from iqm.qiskit_iqm.iqm_provider import IQMBackend
 
 
 class IQMMoveLayout(TrivialLayout):
-    r"""Creates a qubit layout that is valid on the quantum architecture specification of the
-    given IQM backend with regard to the move gate. In more detail, assumes that the move
+    """Creates a qubit layout that is valid on the quantum architecture specification of the
+    given IQM target with regard to the MOVE gate. In more detail, assumes that the MOVE
     operations in the quantum architecture define which physical qubit is the resonator and
-    which is a move qubit, and shuffles the logical indices of the circuit so that they match
+    which is a MOVE qubit, and shuffles the logical indices of the circuit so that they match
     the requirements.
 
     This is required because Qiskit's basic layout algorithm assumes each connection between
     two qubits has the same gates defined.
 
-    Note: This simple version of the mapper only works reliably with a single move qubit
-    and resonator, and only if the circuit contains at least one move gate."""
-
-    def __init__(self, backend: IQMBackend):
-        super().__init__(backend.target)
-        self._backend = backend
+    Note: This simple version of the mapper only works reliably with a single MOVE qubit
+    and resonator, and only if the circuit contains at least one MOVE gate."""
```

Comment:
I guess this needs to be fixed at some point?

Reply:
I wrote a new layout generator that seems to pass the tests. It still assumes that there's just a single resonator.